File: ibexnews-ng/src/app/classes/ibex-company/ibex-company.ts
import {News} from "../news/news";
import {Quote} from "../daily-stocks/quote";

export class IbexCompany {

  constructor(private _queryName: string,
              private _symbol: string,
              private _productionName: string,
              private _cincoDiasTagName: string,
              private _elConfidencialTagName: string,
              private _elEconomistaTagName: string) {
  }

  private _news: News[] = [];
  private _daily! : Quote;


  get daily(): Quote {
    return this._daily;
  }

  set daily(value: Quote) {
    this._daily = value;
  }

  get news(): News[] {
    return this._news;
  }

  set news(value: News[]) {
    this._news = value;
  }

  get queryName(): string {
    return this._queryName;
  }

  set queryName(value: string) {
    this._queryName = value;
  }

  get symbol(): string {
    return this._symbol;
  }

  set symbol(value: string) {
    this._symbol = value;
  }

  get productionName(): string {
    return this._productionName;
  }

  set productionName(value: string) {
    this._productionName = value;
  }

  get cincoDiasTagName(): string {
    return this._cincoDiasTagName;
  }

  set cincoDiasTagName(value: string) {
    this._cincoDiasTagName = value;
  }

  get elConfidencialTagName(): string {
    return this._elConfidencialTagName;
  }

  set elConfidencialTagName(value: string) {
    this._elConfidencialTagName = value;
  }

  get elEconomistaTagName(): string {
    return this._elEconomistaTagName;
  }

  set elEconomistaTagName(value: string) {
    this._elEconomistaTagName = value;
  }
}

File: webscraper-service/src/main/java/com/jasato/webscrapper/webscraper/models/IbexCompany.java
package com.jasato.webscrapper.webscraper.models;

import com.fasterxml.jackson.annotation.*;
import com.sun.istack.*;

import javax.persistence.*;
import javax.persistence.Entity;
import java.util.*;

@Entity
public class IbexCompany {
    @Id
    private String queryName;
    @NotNull
    private String symbol;
    private String productionName;
    private String cincoDiasTagName;
    private String elConfidencialTagName;
    private String elEconomistaTagName;
    @OneToMany( mappedBy = "ibexCompany")
    @JsonIgnore
    private List<News> news;

    public IbexCompany() {
    }

    public IbexCompany(String queryName, String symbol, String name, String cincoDiasTagName, String elConfidencialTagName, String elEconomistaTagName) {
        this.queryName = queryName;
        this.symbol = symbol;
        this.productionName = name;
        this.cincoDiasTagName = cincoDiasTagName;
        this.elConfidencialTagName = elConfidencialTagName;
        this.elEconomistaTagName = elEconomistaTagName;
    }

    public String getSymbol() {
        return symbol;
    }

    public void setSymbol(String symbol) {
        this.symbol = symbol;
    }

    public String getProductionName() {
        return productionName;
    }

    public void setProductionName(String name) {
        this.productionName = name;
    }

    public List<News> getNews() {
        return news;
    }

    public void setNews(List<News> news) {
        this.news = news;
    }

    public String getCincoDiasTagName() {
        return cincoDiasTagName;
    }

    public void setCincoDiasTagName(String cincoDiasTagName) {
        this.cincoDiasTagName = cincoDiasTagName;
    }

    public String getElConfidencialTagName() {
        return elConfidencialTagName;
    }

    public void setElConfidencialTagName(String elConfidencialTagName) {
        this.elConfidencialTagName = elConfidencialTagName;
    }

    public String getElEconomistaTagName() {
        return elEconomistaTagName;
    }

    public void setElEconomistaTagName(String elEconomistaTagName) {
        this.elEconomistaTagName = elEconomistaTagName;
    }

    public String getQueryName() {
        return queryName;
    }

    public void setQueryName(String queryName) {
        this.queryName = queryName;
    }
}
